src/components/OntologyBrowser/store.ts

```diff
@@ -61,6 +72,47 @@ const findAllSameTerms = (
   return sameTermKeys;
 };
 
+export const generateNavTreeFormKey = (
+  phenotypes: string[],
+  selectedPhenotypeInfoTitle: string,
+): TreeNodeLight[] => {
+  if (!phenotypes.length) {
+    return [];
+  }
+
+  if (phenotypes.length === 1) {
+    const leafPheno = phenotypes.pop();
+    const splitPheno = splitHPOTerm(leafPheno || '');
+
+    return [
+      {
+        key: leafPheno,
+        title: hpoTreeTitleFormat(splitPheno, selectedPhenotypeInfoTitle),
+        children: [],
+      } as TreeNodeLight,
+    ];
+  }
+  const rootPheno = phenotypes.pop();
+  const splitPheno = splitHPOTerm(rootPheno || '');
+
+  return [
+    {
+      key: rootPheno,
+      title: hpoTreeTitleFormat(splitPheno, selectedPhenotypeInfoTitle),
+      children: generateNavTreeFormKey(phenotypes, selectedPhenotypeInfoTitle),
+    } as TreeNode,
```

Comment:
line 84-85 and 95-96 are exactly the same. If you use a type/object constructor, this would reduce the logic here.